service/resource_router_test.go

```diff
@@ -0,0 +1,90 @@
+package service
```

Comment:
Did you see https://github.com/giantswarm/aws-operator/blob/bcf155ea823ebf75a562e12faef07ae2d739d35d/service/resource_router_test.go? I am not sure which one is "better" but it would be good to keep these things aligned because we should always try to find some common ground that we can abstract away. This effort gets harder when we build everything from scratch again and when everything is different even though it is doing the same thing.

Reply:
I got the point but for AWS example there is so much instanciation of resrouce that it's not needed by the test.

I would propose to mimic the process but not all the instanciation.

WDYT?

Reply:
When this here is an improvement, fine with me. We do not need to align with the "weaker" implementation. I did not check myself in detail. When you are fine with the tests here lets go with it. We should then just make sure we align in the other projects eventually.

Reply:
Done